src/preferences/dialog/dlgpreflibrary.cpp

```diff
@@ -195,6 +226,21 @@ void DlgPrefLibrary::slotUpdate() {
     m_iOriginalTrackTableRowHeight = m_pLibrary->getTrackTableRowHeight();
     spinBoxRowHeight->setValue(m_iOriginalTrackTableRowHeight);
     setLibraryFont(m_originalTrackTableFont);
+
+    m_iOriginalMaxThreads = m_pconfig->getValue<int>(ConfigKey("[Library]", "MaxAnalysisThreads"));
```

Comment:
The code for calculating this number is duplicated. The preference dialogs should be dumb and should stay in sync with the corresponding components. If you change this code in one place you will most likely forget to update the other location.